### PR TITLE
MAHOUT-1622: MultithreadedBatchItemSimilarities output fix

### PR DESCRIPTION
**Pull request by avati**
Rebased batchSimilarities.patch attached in MAHOUT-1622 and resolved conflicts. Tests pass on laptop and good for merge.

### CODE REVIEW

**Comment by smarthi** [comment]
Patch looks good to ship, would appreciate if u could update the codebase for Java 7. Remove calls to guava collections like Lists, Maps etc and replace them by Java 7 constructs.

**Comment by avati** [comment]
Updated.

File: mr/src/main/java/org/apache/mahout/cf/taste/impl/similarity/precompute/MultithreadedBatchItemSimilarities.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicInteger;
 
-import com.google.common.collect.Lists;
+import java.util.ArrayList;
 import com.google.common.io.Closeables;
```
Comment (smarthi) [inline mr/src/main/java/org/apache/mahout/cf/taste/impl/similarity/precompute/MultithreadedBatchItemSimilarities.java]: U should be able to get rid of Closeables too, and replace that with Java 7 try - resources;  don't worry about that I'll take care of it. JFyi.

Comment (avati) [inline mr/src/main/java/org/apache/mahout/cf/taste/impl/similarity/precompute/MultithreadedBatchItemSimilarities.java]: Argh, I blindly replaced only collections! I could have removed Closeables as well, would be trivial. I'm assuming you will be doing it, based on your previous comment.

Comment (smarthi) [inline mr/src/main/java/org/apache/mahout/cf/taste/impl/similarity/precompute/MultithreadedBatchItemSimilarities.java]: Go ahead and do it if u like, I have a big change coming up which fixes
some of that stuff.

Comment (avati) [inline mr/src/main/java/org/apache/mahout/cf/taste/impl/similarity/precompute/MultithreadedBatchItemSimilarities.java]: Done / pushed.

Comment (avati) [inline mr/src/main/java/org/apache/mahout/cf/taste/impl/similarity/precompute/MultithreadedBatchItemSimilarities.java]: For some reason there is a delay in GitHub reflecting branch changes in PR.

Comment (smarthi) [inline mr/src/main/java/org/apache/mahout/cf/taste/impl/similarity/precompute/MultithreadedBatchItemSimilarities.java]: I am seeing github issues too, consistently been getting RPC timeouts for
the past 30 mins.

**Comment by avati** [comment]
Creating fresh PR to reflect updates

**Pull request closed without merging**